controllers/nicclusterpolicy_controller.go

```diff
@@ -249,20 +253,39 @@ func (r *NicClusterPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	}
 	r.stateManager = stateManager
 
-	builder := ctrl.NewControllerManagedBy(mgr).
+	ctl := ctrl.NewControllerManagedBy(mgr).
 		For(&mellanoxv1alpha1.NicClusterPolicy{}).
+		WithOptions(controller.Options{MaxConcurrentReconciles: 1}).
 		// Watch for changes to primary resource NicClusterPolicy
 		Watches(&source.Kind{Type: &mellanoxv1alpha1.NicClusterPolicy{}}, &handler.EnqueueRequestForObject{})
 
-	// Watch for changes to secondary resource DaemonSet and requeue the owner NicClusterPolicy
+	// we always add object with a same(static) key to the queue to reduce
+	// reconciliation count
+	qHandler := func(q workqueue.RateLimitingInterface) {
+		q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
```

Comment:
wont reconcile loop fail in this case ? as we are trying to get a non existent resource ?

nic cluster policy is cluster scope (no namespace)

Reply:
also you are calling qhandler exactly once, so you can just put the logic in UpdateFunc at L#273 below directly

Reply:
> also you are calling qhandler exactly once, so you can just put the logic in UpdateFunc at L#273 below directly

IMO, it's more readable but I can change if needed